### PR TITLE
refactor: [ST-2960] Fix deprecations

### PR DESCRIPTION
**Pull request by bryanzerrudo**
### Purpose/goal of PR (Issue #)

https://wovnio.atlassian.net/browse/ST-2960

Changes:
- Fixes the following deprecations:
  - "Deprecate ${} string interpolation" (see [PHP RFC](https://wiki.php.net/rfc/deprecate_dollar_brace_string_interpolation))
  - "Deprecate dynamic properties" (see [PHP RFC](https://wiki.php.net/rfc/deprecate_dynamic_properties))
- Adds tests for PHP 8.1 to 8.3

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -20,6 +20,7 @@ commands:
           at: ~/project
       - setup_remote_docker:
           docker_layer_caching: true
+          version: docker25
```
Comment (bryanzerrudo) [inline .circleci/config.yml:23]: See also: [Docker docs](https://docs.docker.com/engine/deprecated/#pushing-and-pulling-with-image-manifest-v2-schema-1)

**Review by bstewart00 — APPROVED** [review]

**Review by bstewart00 — APPROVED** [review]
since production code was affected, bump patch version

**Comment by bryanzerrudo** [comment]
> since production code was affected, bump patch version

I was considering not making a release this time. WDYT?

**Comment by Soichiro725** [comment]
> > since production code was affected, bump patch version
> 
> Since there are no behavior changes, I was considering not making a release this time. WDYT?

I think we should bump version, when we change production code. So, we should make new release in this case.

File: test/unit/StoreTest.php [hunk 2]
```diff
@@ -28,8 +28,6 @@ private function getMockAndRegister($originalClassName, $methods)
         $mockObject = $builder->getMock();
         if (method_exists($this, 'registerMockObject')) {
             $this->registerMockObject($mockObject);
-        } else {
-            $this->mockObjects[] = $mockObject;
```
Comment (bryanzerrudo) [inline test/unit/StoreTest.php:32]: Unused variable.

**Review by iwai-wovn — APPROVED** [review]

**Pull request merged**